### PR TITLE
Add Dockerfile and make docker target

### PR DESCRIPTION
**Pull request by kba**

### CODE REVIEW

**Comment by kba** [comment]
It is ready.

@vahidrezanezhad @michalbubula this adds a Dockerfile, so we can run eynollah in docker, podman or kubernetes. It does not change anything in the code and should be compatible with all branches, so merging should have no adverse effects. With the default base image, there won't be support for GPU though but that can be rectified later (we do have cuda-tf2 core base images that should enable that but since we cannot use GPU in our cluster right now, it's low prio for us atm)

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    org.label-schema.vcs-ref=$VCS_REF \
+    org.label-schema.vcs-url="https://github.com/qurator-spk/eynollah" \
+    org.label-schema.build-date=$BUILD_DATE
+
+ENV DEBIAN_FRONTEND=noninteractive
+ENV PYTHONIOENCODING=utf8
+ENV XDG_DATA_HOME=/usr/local/share
+
+WORKDIR /build-eynollah
+COPY qurator/ ./qurator
```
Comment (cneud) [inline Dockerfile:17]: > COPY qurator/ ./qurator

Is this still needed/correct after changing the package layout?

Comment (kba) [reply Dockerfile]: Argh, I wrote this against the outdated version in Gitlab. Thanks for spotting. I'll update.

**Review by kba — COMMENTED** [review]
Updated and added a smoke test for make docker to GHA

**Review by cneud — APPROVED** [review]

**Pull request merged**